### DISCUSSION TITLE
With UniFi OS - Dream Machines 4.x.x. in EA, does nspawn-container still work?

### DISCUSSION THREAD

**Discussion by waffles0042** [unifi-utilities/unifi-common#604] [General]
[Unify recently released 4.0.3 to EA.](https://community.ui.com/releases/UniFi-OS-Dream-Machines-4-0-3/90b24b90-8be8-4987-8d47-eb408eb07e45?page=1) . I was wondering if the [method for creating a custom container on UnifiOS 3.x+ and consequently installing things like PiHole still works](https://github.com/unifi-utilities/unifios-utilities/tree/main/nspawn-container). 
Has anyone tried it yet and could share their findings? TIA!

**Comment by waffles0042** [comment]
To answer my own question, in case s.o. is wondering:
I just upgraded my UDM-Pro to 4.0.6 Release Candidate and everything (e.g. PiHole) still seems to have survived the upgrade and is running fine.

**Comment by hart-NTP** [comment]
I've been using the original on boot script with https://github.com/telnetdoogie/UDMP-ipv6 on a UDM Pro and just upgraded to the final UniFi OS 4.06 (not EA).

I believe the on boot stuff didn't survive the upgrade,  Perhaps it would have if I used the newer on boot repo instead, I didn't know about it.  Reinstalling on boot using pipe download to sh fixed it, and after a reboot my IPv6 tunnel came back up.

**Comment by robigan** [comment]
On Boot works but ultimately failed as my scripts failed. The only script that failed is `10-setup-network.sh` as this line
https://github.com/unifi-utilities/unifios-utilities/blob/6c899cbc2a2dd6e3d1fd77308c76a813680c28df/nspawn-container/scripts/10-setup-network.sh#L60-L64
produces the following error:
`grep: /run/dnsmasq.conf.d/custom.conf: No such file or directory`

**Reply by m4n63** [reply]
In my experience, that message is just the if clause failing, and while it does spit out the message, the script should still exit with a success as long as the other part are ok.

**Reply by robigan** [reply]
> In my experience, that message is just the if clause failing, and while it does spit out the message, the script should still exit with a success as long as the other part are ok. 

Realized this later. Then yeah all works
